### PR TITLE
Removed the Go path link from the ./eventing/triggers/README.md

### LINKED ISSUE #3309: Remove links to Go code
We have some pages that link to Go code (especially Go types).
Most of the time while reading the documentation is not expected to see that a link brings you to a low-level representation of the resource you're learning.
In addition to that, users might not be familiar with Go and they should not think that Go is required for learning or using Knative.

## Expected Behavior

No link to Go code.

## Additional information

We should find all links to Go code, and then for each of them, we can replace with an example using YAML or the `kn` CLI

_If you want to tackle a specific page or link, please comment on this issue, so that there is no duplicate effort from multiple people_

/good-first-issue

### PR DESCRIPTION
**Pull request by Yashks1994**
Fixes #3309 - Remove links to Go code

## Proposed Changes 

- Removed the Go path link for  Destination

### CODE REVIEW

File: docs/eventing/triggers/README.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 A Trigger represents a desire to subscribe to events from a specific Broker.
 
-The `subscriber` value must be a [Destination](https://pkg.go.dev/knative.dev/pkg/apis/duck/v1#Destination).
+The `subscriber` value must be a Destination.
```
Comment (pierDipi) [inline docs/eventing/triggers/README.md:3]: In the other PR, I suggested a replacement for this.

Comment (Yashks1994) [reply docs/eventing/triggers/README.md:3]: I have added the Destination yaml but my point is comment yaml the comment is mentioned for DeadLinkSink.
# DeadLetterSink is the sink receiving event that could not be sent to
# a destination.

So, is it still relevant to keep it?

**Comment by Yashks1994** [comment]
@pierDipi as file has been removed from the main branch, so after resolving the conflict the file gets removed.
Should I keep it open or need to close.

**Review by pierDipi — COMMENTED** [review]
There are unrelated changes to this PR.

**Comment by pierDipi** [comment]
> @pierDipi as file has been removed from the main branch, so after resolving the conflict the file gets removed.
Should I keep it open or need to close.

Let's close this then and find out what's left once we sort out the other PRs in https://github.com/knative/docs/issues/3309

**Pull request closed without merging**